src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetHash.cs

```diff
@@ -125,24 +129,36 @@ protected override void ProcessRecord()
             }
         }
 
-        private byte[] ComputeHash(Stream stream)
+        private async Task<byte[]> ComputeHashAsync(Stream stream, CancellationToken cancellationToken)
         {
             switch (Algorithm)
             {
                 case HashAlgorithmNames.SHA1:
-                    return SHA1.HashData(stream);
+                    return await SHA1.HashDataAsync(stream, cancellationToken);
```

Comment:
```suggestion
                    return await SHA1.HashDataAsync(stream, cancellationToken).ConfigureAwait(false);
```

Without this we would potentially dead lock when winforms (or something else that adds a sync context) is active

Reply:
I updated the code to just return the ValueTask directly, without awaiting. That way, the code should be safe even without `ConfigureAwait`, since it does not affect `GetResult`, right? (according to https://devblogs.microsoft.com/dotnet/configureawait-faq/, section "I’m using GetAwaiter().GetResult(). Do I need to use ConfigureAwait(false)?")

Reply:
No, we should await. Please revert.

Reply:
Why?

Reply:
If an asynchronous method is used, you need to use await to get the result, as was the case in the first commit.

Reply:
you don't *need* to if you're just returning the task and not actually waiting for it inside the method